homeassistant/data_entry_flow.py

```diff
@@ -298,6 +308,34 @@ async def async_init(
 
         return result
 
+    def _set_error_on_path(
+        self,
+        schema_errors: dict[str, Any],
+        error_message: str,
+        error_path: list[str | vol.Marker] | None,
+        data_schema: vol.Schema,
+    ) -> dict[str, Any]:
+        """Set an error in schema_errors with the correct path in the schema."""
+        schema = data_schema.schema
+        if not error_path or (path_part := error_path[0]) not in schema:
+            raise ValueError("Could not find path in schema")
+
+        path_part_str = str(path_part)
+        result: str | dict[str, Any]
+        if len(error_path) == 1:
+            # last path
+            result = error_message
+        else:
+            result = self._set_error_on_path(
+                schema_errors.get(path_part_str, {}),
+                error_message,
+                error_path[1:],
+                schema[path_part],
+            )
```

Comment:
Currently, nested schemas are not supported as `voluptuous_serialize` has no support for them.

Should we leave the error handling for nested ones or should we remove the handling for it and handle that error like the ones with an unknown path?
See `test_flow_with_multiple_schema_errors_nested`

Reply:
YAGNI, remove code which is not needed, but document the limitation in the docstring